data_structures/src/transaction.rs

```diff
@@ -0,0 +1,430 @@
+pub struct CommitTransactionBody {
+    // Inputs
+    // TODO: Discussion about collateral
+    //pub collateral: Vec<Input>, // ValueTransferOutput
+    pub dr_pointer: Hash, // DTTransaction hash
+    // Outputs
+    pub commitment: Hash,
+    // Proofs
+    pub repoe: Hash, // Proof of elegibility for this pkh and this data request
+
+    #[protobuf_convert(skip)]
+    #[serde(skip)]
+    hash: MemoHash,
+}
+
+impl CommitTransactionBody {
+    /// Creates a new value transfer transaction body.
+    pub fn new(dr_pointer: Hash, commitment: Hash, repoe: Hash) -> Self {
+        CommitTransactionBody {
+            dr_pointer,
+            commitment,
+            repoe,
+            hash: MemoHash::new(),
+        }
+    }
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert)]
+#[protobuf_convert(pb = "witnet::RevealTransaction")]
+pub struct RevealTransaction {
+    pub body: RevealTransactionBody,
+    pub signatures: Vec<KeyedSignature>,
+}
+
+impl RevealTransaction {
+    /// Creates a new value transfer transaction body.
+    pub fn new(body: RevealTransactionBody, signatures: Vec<KeyedSignature>) -> Self {
+        RevealTransaction { body, signatures }
+    }
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert)]
+#[protobuf_convert(pb = "witnet::RevealTransactionBody")]
+pub struct RevealTransactionBody {
+    // Inputs
+    pub dr_pointer: Hash, // DTTransaction hash
+    // Outputs
+    pub reveal: Vec<u8>,
+    pub pkh: PublicKeyHash, // where to receive reward
+
+    #[protobuf_convert(skip)]
+    #[serde(skip)]
+    hash: MemoHash,
+}
+
+impl RevealTransactionBody {
+    /// Creates a new value transfer transaction body.
+    pub fn new(dr_pointer: Hash, reveal: Vec<u8>, pkh: PublicKeyHash) -> Self {
+        RevealTransactionBody {
+            dr_pointer,
+            reveal,
+            pkh,
+            hash: MemoHash::new(),
+        }
+    }
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert)]
+#[protobuf_convert(pb = "witnet::TallyTransaction")]
+pub struct TallyTransaction {
+    // Inputs
+    pub dr_pointer: Hash, // DTTransaction hash
+    // Outputs
+    pub tally: Vec<u8>,
+    pub outputs: Vec<ValueTransferOutput>, // Witness rewards
+
+    #[protobuf_convert(skip)]
+    #[serde(skip)]
+    hash: MemoHash,
+}
+
+impl TallyTransaction {
+    /// Creates a new value transfer transaction body.
+    pub fn new(dr_pointer: Hash, tally: Vec<u8>, outputs: Vec<ValueTransferOutput>) -> Self {
+        TallyTransaction {
+            dr_pointer,
+            tally,
+            outputs,
+            hash: MemoHash::new(),
+        }
+    }
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert)]
+#[protobuf_convert(pb = "witnet::MintTransaction")]
+pub struct MintTransaction {
+    pub epoch: Epoch,
+    pub outputs: Vec<ValueTransferOutput>,
```

Comment:
I think the mint transaction should only have one output

```suggestion
    pub output: ValueTransferOutput,
```

Reply:
I will add a new Issue for that, because now most of the methods which update utxos use a `Vec<ValueTransferOutput>`

Reply:
Thanks! For reference, issue number is #650